### PR TITLE
QOL-8720 add workflow to make pr's for latest

### PR DESCRIPTION
**Pull request by duttonw**
Hopefully this works.

It should use our npm token to gain access to nexus, pull formio premium and formiojs latest and push to lib folder and create a pr

also have script to auto make change log file.

### CODE REVIEW

File: .github/workflows/formioLibUpdate.js.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          node-version: 16
+          cache: 'npm'
+          cache-dependency-path: perm/package-lock.json
+          registry-url: 'https://registry.npmjs.org'
+      - name: List versions
+        working-directory: ./perm
+        run: | # List versions
+          npm view formiojs versions
+          npm view @formio/premium versions
+        env:
```
Comment (ThrawnCA) [inline .github/workflows/formioLibUpdate.js.yml:27]: `env` can be defined at a higher level instead of being repeated.

**Review by ThrawnCA — CHANGES_REQUESTED** [review]
Build is broken due to Nexus authentication failure.

File: .github/workflows/formioLibUpdate.js.yml (hunk 1 above)
Comment (duttonw) [reply .github/workflows/formioLibUpdate.js.yml:27]: umm, i did not 'alter' those workflow's which failed. What was run was 
https://github.com/qld-gov-au/formio/actions/runs/2193583748/workflow which had no registry changes.

Comment (ThrawnCA) [reply .github/workflows/formioLibUpdate.js.yml:27]: Sure, the `env` suggestion is unrelated to the build failure. It's just a suggestion. I'm unsure where the authentication failure is coming from.

**Comment by devonpis** [comment]
maybe try using cli `npm set registry=https://nexus.tools.services.qld.gov.au/nexus/repository/npm_all/ && npm set _auth=${secret...}` instead for the 401 error in building step

**Comment by duttonw** [comment]
Am trying https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#use-private-packages 

Use private packages
```
steps:
- uses: actions/checkout@v3
- uses: actions/setup-node@v3
  with:
    node-version: '14.x'
    registry-url: 'https://registry.npmjs.org'
# Skip post-install scripts here, as a malicious
# script could steal NODE_AUTH_TOKEN.
- run: npm install --ignore-scripts
  env:
    NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
# `npm rebuild` will run all those post-install scripts for us.
- run: npm rebuild && npm run prepare --if-present
```

**Comment by devonpis** [comment]
> Am trying https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#use-private-packages
> 
> Use private packages
> 
> ```
> steps:
> - uses: actions/checkout@v3
> - uses: actions/setup-node@v3
>   with:
>     node-version: '14.x'
>     registry-url: 'https://registry.npmjs.org'
> # Skip post-install scripts here, as a malicious
> # script could steal NODE_AUTH_TOKEN.
> - run: npm install --ignore-scripts
>   env:
>     NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
> # `npm rebuild` will run all those post-install scripts for us.
> - run: npm rebuild && npm run prepare --if-present
> ```

probably have to rename the env variable name?
ie.

```
env:
  _auth=${{ secrets.NEXUSREADONLY2NPMTOKEN }}
```

**Review by ThrawnCA — APPROVED** [review]

**Review by devonpis — APPROVED** [review]

**Pull request merged**